### DISCUSSION TITLE
Does Lima work with Vagrant? (Is there a Vagrant Provider for Lima?)

### DISCUSSION THREAD

**Discussion by techieshark** [lima-vm/lima#1271] [Q&A] [answered]
Given that VirtualBox doesn't work on M1 Macs anymore, and Lima seems like a good Docker alternative that is performant & open source, I'm looking into Lima.

But previously I had used VirtualBox configured by Vagrant (with [Trellis](https://roots.io/trellis/)) and I'm wondering if I can have Vagrant set up / control Lima the same way it can for VirtualBox or Docker. 

So: does anyone know if there is a [Vagrant Provider](https://developer.hashicorp.com/vagrant/docs/providers) for Lima? 

I see it's possible to create a [Custom Provider](https://developer.hashicorp.com/vagrant/docs/plugins/providers) for Vagrant, but I thought I'd check if anyone has already done such a thing for Lima.

**Comment by afbjorklund** [comment]
I have mostly _translated_ my old Vagrantfile (ruby) into Lima (yaml), but it sounds like a good addition...

There are some old warts like the "known ssh key" (instead of cloud-init), but I guess that it should work ?

Perhaps it would make more sense to look towards Vagrant3 ? (since it is using Go, rather than Ruby)

https://www.hashicorp.com/blog/toward-vagrant-3-0

**Reply by darkn3rd** [reply]
That might actually be a good idea, a Vagrant<->Lima translator. ;-)

**Reply by AkihiroSuda** [reply, marked as answer]
> That might actually be a good idea, a Vagrant<->Lima translator. ;-)

AI models can easily translate them.
e.g.,
- https://github.com/containerd/containerd/pull/13728

**Comment by aurynn** [comment]
I'm interested in either a provider or better documentation on mimicking Vagrant with Lima (or Colima), as the vagrant-go rewrite was just [announced as dead](https://github.com/hashicorp/vagrant/issues/12819#issuecomment-2738619021).

**Reply by afbjorklund** [reply]
On the other hand, VirtualBox now also works on ARM Macs (since 7.1). So there is probably no need for a vagrant provider, but documentation could be good.

**Reply by neurogenesis** [reply]
Using VirtualBox can expose you to the whims of Oracle. A few years ago they sent out threatening legal notices to enterprises, claiming software license violations to scare companies into paying for the open source software. The fine print involved Oracle's proprietary "Extension Pack", whether installed or not. VirtualBox itself is free and open under the GPLv3 license, allowing modification and distribution. It was originally created by InnoTek Systemberatung GmbH, acquired by Sun Microsystems in 2008, and in turn acquired by Oracle in 2010.

Alternatives to the commercial providers Oracle, VMware, and Parallels would certainly be welcome. It would combine the automation capabilities of vagrant with the speed of lima.

**Reply by afbjorklund** [reply]
Vagrant is not Open Source anymore (since version 2.3.7), but there were some attempts to document "Lima" vs "Vagrant" mode

* https://github.com/lima-vm/lima/discussions/1749
* https://github.com/lima-vm/lima/issues/3749

**Reply by darkn3rd** [reply]
I think Vagrant 2.3.7 needs to be forked.  Hashicorp doesn't seem to be maintaining it.  The go rewrite likely did not happen in that (1) Vagrant is really well written in general, (2) the provisioners cover CMs that have tiny markets, (3) post-container world, interest may be limited, at least commercially, it would thrive better in pure open source, and making solutions around it, such as cloud-terraform-provisioner plugins with multi-machine for AWS ASGs, Azure VMSSs, GCP MIGs, or containers on K8S or Nomad from Vagrant.
